src/ott/initializers/linear/initializers.py

```diff
@@ -313,3 +332,108 @@ def body_fn(i: int, f: jnp.ndarray) -> jnp.ndarray:
     return f.at[i].set(new_f)
 
   return jax.lax.fori_loop(0, len(f), body_fn, f)
+
+
+@functools.partial(jax.jit, static_argnums=(3,))
+def subsample(
+    rng: jax.random.PRNGKey, array: jnp.ndarray, weight: jnp.ndarray,
+    subsample_n: int
+):
+  """Subsample point cloud to uniform weighted points.
+
+  Args:
+    rng: random number generator.
+    array: point cloud, array of size N x d.
+    weight: array of size N.
+    subsample_n: number of points to subsample.
+
+  Returns:
+    subsampled point cloud, array of size subsample_n x d.
+  """
+  N = array.shape[0]
+  indices = jax.random.choice(
+      key=rng, a=N, shape=(subsample_n,), replace=True, p=weight, axis=0
+  )
+  return array[indices]
+
+
+@jax.tree_util.register_pytree_node_class
+class SubsampleInitializer(DefaultInitializer):
+  """Subsample initializer :cite:`thornton2022rethinking:22`.
+
+  Subsample each point cloud, then compute closed from
+  Sinkhorn potential between subsampled approximations. Use this potential
+  to initialize Sinkhorn potentials/scalings.
+
+  Args:
+    subsample_n: number of points to subsample from each point cloud.
+  """
+
+  def __init__(
+      self,
+      subsample_n: int,
```

Comment:
what's the best was of adding sinkhown kwargs? Need to add as an attribute of class, then pass to sinkhorn in method later on.

Is this the best way?
 ```
def init(self, ..., **kwargs):
  ...
 self.kwargs =kwargs 

def init_dual_a(...):
  sinkhorn.solve(geom, **self.kwargs)
```

Reply:
Yes, this is one option. Other one would be to directly pass the solver and let caller construct it.
I think I like this approach because it's more general than the current one (doesn't limit us to Sinkhorn solvers in the future), but it's a bit less convenient. Wdyt?

Reply:
I think the initialiser just works for the sinkhorn solver right now. If there's a strong use case it be worth looking into in the future. 

In order to pass solver we would need a fixed interface or a solver base class, which I do not believe is there yet.

Reply:
Let's keep it as it is now.